File: lambdas/origin-router-lambda.js
'use strict';

exports.handler = (event, context, callback) => {
    console.info("Event Received " + JSON.stringify(event));

    let keysList = [
        'www',
        'justinlmartin',
        'portfolio',
        'test-site',
        'peaks-valleys',
        'creative-coding',
        'spaceblobman',
        'sunrise-sunset',
        'blackjack',
        'sunflower-synth',
        'sampampulancer'
    ]

    const request = event.Records[0].cf.request
    let bucketDomain = event.Records[0].cf.request.origin.s3.domainName

    let routingUrl = event.Records[0].cf.request.headers.host[0].value;
    let subdomain = routingUrl.split('.')[0]
    let site = subdomain.replace('-dev', ' ')

    console.info('Routing: ', routingUrl, subdomain, site)

    if (!keysList.includes(site)) {
        console.error('Incorrect subdomain. Routing to default root object.')
    } else if ( site === 'www' || site === 'portfolio' || site === 'justinlmartin') {
        console.info('Routing to portfolio')
        request.origin.s3.path = `/portfolio`
    } else {
        console.info(`Routing to ${site}`)
        request.origin.s3.path = `/${site}`
    }

    request.headers['host'] = [{ key: 'Host', value: bucketDomain}];
    console.log('Updated request to:' + JSON.stringify(request));
    callback(null, request)
}

File: lambdas/origin-router-lambda/origin-router-lambda.py
import logging

logger = logging.getLogger()
logger.setLevel(logging.INFO)


def handler(event, context):
    # test comment
    logger.info('Received event: {event}'.format(event=event))

    keys_list = [
        'www',
        'justinlmartin',
        'coding-portfolio',
        'qr',
        'landing-page',
        'test-site',
        'peaks-valleys',
        'creative-coding',
        'spaceblobman',
        'sunrise-sunset',
        'blackjack',
        'sunflower-synth',
        'sampampulancer',
        'coding-portfolio-dev',
        'qr-dev',
        'landing-page-dev',
        'test-site-dev',
        'peaks-valleys-dev',
        'creative-coding-dev',
        'spaceblobman-dev',
        'sunrise-sunset-dev',
        'blackjack-dev',
        'sunflower-synth-dev',
        'sampampulancer-dev'
    ]

    request = event['Records'][0]['cf']['request']
    bucket_domain = event['Records'][0]['cf']['request']['origin']['s3']['domainName']
    routing_url = event['Records'][0]['cf']['request']['headers']['host'][0]['value']
    subdomain = routing_url.split('.')[0]
    # site = subdomain.replace('-dev', '')

    logger.info('Routing: {url}, {subdomain}'.format(url=routing_url, subdomain=subdomain))

    if subdomain not in keys_list:
        logger.error('Incorrect subdomain. Re-Routing to default root object.')
    elif subdomain == 'www' or subdomain == 'justinlmartin' or subdomain == 'qr':
        logger.info('Re-Routing to prod landing page')
        request['origin']['s3']['path'] = '/landing-page'
    elif subdomain == 'qr-dev':
        logger.info('Re-Routing to dev landing page')
        request['origin']['s3']['path'] = '/landing-page-dev'
    else:
        logger.info('Routing to {subdomain}'.format(subdomain=subdomain))
        request['origin']['s3']['path'] = '/{subdomain}'.format(subdomain=subdomain)

    request['headers']['host'] = [{
        'key': 'Host',
        'value': bucket_domain
    }]
    logger.info('Updated request to: {request}'.format(request=request))
    return request

